athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/security/CachableSecretsManager.java

```diff
@@ -80,7 +80,7 @@ public String resolveSecrets(String rawString)
             String nextSecret = m.group(1);
             Matcher m1 = NAME_PATTERN.matcher(nextSecret);
             m1.find();
-            result = result.replace(nextSecret, getSecret(m1.group(1)));
+            result = result.replace(nextSecret, Matcher.quoteReplacement(getSecret(m1.group(1))));
```

Comment:
Looks like String.replace(...) already calls Matcher.quoteReplacement(...). I was mistaken about this when I originally suggested that we might need to apply the fix here too.